### PR TITLE
Proposition of save as_dict in get_theme but with change default behaviour in future

### PR DESCRIPTION
**Pull request by Czaki**
# Description

As qss needs colors in `rgb(x,y,z)` then saving option to convert Theme to old-style dictionary is necessary. It could be done in `get_theme` function, in a separate function or as method of `Theme`class. To reduce changes in depending codebase I suggest keeping `as_dict` option in the code, and only change its default value in a future release. 

I also use `None` to distinguish user-provided value and the default one.  

## Type of change

- [ ] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

**Review by tlambert03 — APPROVED** [review]
fine by me!

File: napari/utils/theme.py [hunk 1]
```diff
@@ -226,6 +207,27 @@ def get_theme(name, as_dict=True):
                 themes=available_themes(),
             )
         )
+    theme = _themes[name]
+    _theme = theme.copy()
+    if as_dict is None:
+        warnings.warn(
+            trans._(
+                "Themes were changed to use evented model with Pydantic's color type rather than the `rgb(x, y, z)`."
+                "The `as_dict=True` option will be changed to `as_dict=False` in 0.X.X",
```
Comment (sofroniewn) [inline napari/utils/theme.py]:
```suggestion
                "The `as_dict=True` option will be changed to `as_dict=False` in 0.4.15",
```

**Review by sofroniewn — APPROVED** [review]
Would you still like this to merge @Czaki or is it no longer needed?

I'm fine with it going in, so if you still want it I can merge tomorrow. Thanks!

**Comment by Czaki** [comment]
> Would you still like this to merge @Czaki or is it no longer needed?

Yes. I still need to set the stylesheet in PartSeg to have one style across the application.

**Pull request merged**